indico/modules/events/editing/models/revisions.py

```diff
@@ -60,9 +60,14 @@ def _make_state_check():
                f_accepted=FinalRevisionState.accepted)).strip()
 
 
+def _make_reviewed_dt_check():
+    return '((final_state = {FinalRevisionState.none}) OR (reviewed_dt IS NOT NULL))'
+
+
 class EditingRevision(RenderModeMixin, db.Model):
     __tablename__ = 'revisions'
     __table_args__ = (db.CheckConstraint(_make_state_check(), name='valid_state_combination'),
+                      db.CheckConstraint(_make_reviewed_dt_check(), name='reviewed_dt_set_when_final_state'),
```

Comment:
the condition could be inlined here; the only reason for the other function is because it's much more complex.
but if it's more readable like this (wrapped strings are ugly!) feel free to keep it

Reply:
I'd keep it as is, I think it reads better :)